### PR TITLE
fix: install-npm.js fails to run npm run build due to type error

### PR DESCRIPTION
**Pull request by rerorero**
`node install-npm.js` in the current main branch doesn't work due to tsc error.
e.g. https://github.com/appium/appium-chromedriver/actions/runs/4403909094/jobs/7720201682

After letting it show the error details with `util.inspect(e)`, npm install failed like this:
```
npm install
npm WARN config init.author.name Use `--init-author-name` instead.
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'semantic-release@20.1.1',
npm WARN EBADENGINE   required: { node: '>=18' },
npm WARN EBADENGINE   current: { node: 'v16.15.1', npm: '8.11.0' }
npm WARN EBADENGINE }

> appium-chromedriver@5.3.0 postinstall
> node install-npm.js

[23:50:48] The Chromedriver install script cannot be found at '/Users/rerorero/go/src/github.com/rerorero/appium-chromedriver/build/lib/install.js'. Building appium-chromedriver package

/Users/rerorero/go/src/github.com/rerorero/appium-chromedriver/node_modules/teen_process/lib/exec.js:129
        let err = new Error(`Command '${rep}' exited with code ${code}`);
                  ^
Error: appium-chromedriver package cannot be built: Error: Command 'npm run build' exited with code 1
    at ChildProcess.<anonymous> (/Users/rerorero/go/src/github.com/rerorero/appium-chromedriver/node_modules/teen_process/lib/exec.js:129:19)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5) {
  stdout: '\n' +
    '> appium-chromedriver@5.3.0 build\n' +
    '> tsc -b\n' +
    '\n' +
    "lib/utils.js(134,62): error TS2345: Argument of type '{}' is not assignable to parameter of type 'DriverOpts<Readonly<Record<string, Constraint>>>'.\n" +
    "  Type '{}' is missing the following properties from type 'ServerArgs': address, allowCors, allowInsecure, basePath, and 26 more.\n",
  stderr: 'npm WARN config init.author.name Use `--init-author-name` instead.\n',
  code: 1
}
    at main (/Users/rerorero/go/src/github.com/rerorero/appium-chromedriver/install-npm.js:41:13)
npm ERR! code 1
npm ERR! path /Users/rerorero/go/src/github.com/rerorero/appium-chromedriver
npm ERR! command failed
npm ERR! command sh -c node install-npm.js

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/rerorero/.npm/_logs/2023-03-14T14_50_47_686Z-debug-0.log
```

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -131,6 +131,9 @@ const getOsInfo = _.memoize(
   }
 );
 
+// @ts-expect-error
+// error TS2345: Argument of type '{}' is not assignable to parameter of type 'DriverOpts<Readonly<Record<string, Constraint>>>'
+// Type '{}' is missing the following properties from type 'ServerArgs': address, allowCors, allowInsecure, basePath, and 26 more.
```
Comment (rerorero) [inline lib/utils.js:136]: This PR just ignores the tsc error as a workaround but probably I think it'd be better to expose [generateDriverLogPrefix](https://github.com/appium/appium/blob/943dce5eec4c047073ef9dfcb97da9ccd6be9459/packages/base-driver/lib/basedriver/helpers.js#L596) function.

File: install-npm.js [hunk 2]
```diff
@@ -37,7 +38,7 @@ async function main() {
     try {
       await exec(npmCommand, ['run', 'build'], {logger: log, cwd: __dirname});
     } catch (e) {
-      throw new Error(`appium-chromedriver package cannot be built: ${e.stderr || e.message}`);
+      throw new Error(`appium-chromedriver package cannot be built: ${util.inspect(e)}`);
```
Comment (mykola-mokhnach) [inline install-npm.js:41]: why this change is necessary? why actually want to display stderr there

Comment (mykola-mokhnach) [reply install-npm.js:41]: maybe `${('stderr' in e && e.stderr) || ('message' in e && e.message)}` would work

Comment (rerorero) [reply install-npm.js:41]: Because stderr doesn't have enough details in this case and I just found it shows everything this way.
```
  stdout: '\n' +
    '> appium-chromedriver@5.3.0 build\n' +
    '> tsc -b\n' +
    '\n' +
    "lib/utils.js(134,62): error TS2345: Argument of type '{}' is not assignable to parameter of type 'DriverOpts<Readonly<Record<string, Constraint>>>'.\n" +
    "  Type '{}' is missing the following properties from type 'ServerArgs': address, allowCors, allowInsecure, basePath, and 26 more.\n",
  stderr: 'npm WARN config init.author.name Use `--init-author-name` instead.\n',
```

Comment (mykola-mokhnach) [reply install-npm.js:41]: what is an example of the `inspect` output?

Comment (mykola-mokhnach) [reply install-npm.js:41]: or maybe then include err.stdout there before stderr?

Comment (rerorero) [reply install-npm.js:41]: I put the output of inspect() in the PR description.
Including stdout also should be fine

**Comment by mykola-mokhnach** [comment]
Seems like tests still need some fixes

**Review by mykola-mokhnach — APPROVED** [review]

**Comment by rerorero** [comment]
rebased with master branch and fixed the lint error
